File: modules/post/server-scala/src/main/scala/model/EditCommentInput.scala
package model

case class EditCommentInput(id: Int, content: String, postId: Int)

object EditCommentInput {

  implicit def inputToComment(input: EditCommentInput): Comment = {
    Comment(id = Some(input.id), content = input.content, postId = input.postId)
  }
}

File: modules/post/server-java/src/main/java/graphql/model/EditCommentInput.java
package graphql.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EditCommentInput {

    private Integer id;

    /**
     * An id of the related post. This is necessary to update the related post with subscription mechanism.
     */
    private Integer postId;

    private String content;
}
